core/Tracker/TableLogAction.php

```diff
@@ -207,17 +203,19 @@ public static function getIdActionFromSegment($valueToMatch, $sqlField, $matchTy
             $sql = self::getSelectQueryWhereNameContains($matchType, $actionType);
         }
 
-
-        $cache = StaticContainer::get('Piwik\Tracker\TableLogAction\Cache');
-        return $cache->getIdActionFromSegment($valueToMatch, $sql);
+        return array(
+            // mark that the returned value is an sql-expression instead of a literal value
+            'SQL' => $sql,
+            'bind' => $valueToMatch,
+        );
     }
 
     /**
      * @param $segmentName
      * @return int
      * @throws \Exception
      */
-    private static function guessActionTypeFromSegment($segmentName)
+    public static function guessActionTypeFromSegment($segmentName)
```

Comment:
this can be reverted
```suggestion
    private static function guessActionTypeFromSegment($segmentName)
```